pkg/scheduler/algorithm/priorities/metadata.go

```diff
@@ -123,32 +123,58 @@ func getSelector(pod *v1.Pod, sl corelisters.ServiceLister, cl corelisters.Repli
 		}
 	}
 
-	if rcs, err := cl.GetPodControllers(pod); err == nil {
-		for _, rc := range rcs {
-			labelSet = labels.Merge(labelSet, rc.Spec.Selector)
+	rcPod, rsPod, stsPod := false, false, false
+	for _, ref := range pod.OwnerReferences {
+		switch ref.Kind {
+		case "ReplicationController":
+			if ref.APIVersion == "v1" {
```

Comment:
do all replication controller pods have an owner reference with this kind/apiversion (even historic ones?)

Reply:
> do all replication controller pods have an owner reference with this kind/apiversion (even historic ones?)

I guess it is.

Reply:
> I guess it is.

we need to verify this by looking at the replication controller behavior back to 1.14